diff --git builds/android/android_build_helper.sh builds/android/android_build_helper.sh
index 6263f70..d0b24c3 100755
--- builds/android/android_build_helper.sh
+++ builds/android/android_build_helper.sh
@@ -202,6 +202,10 @@ function _android_build_opts_process_cxx_stl {
             LDFLAGS+=" -L${ANDROID_NDK_ROOT}/sources/cxx-stl/gnu-libstdc++/4.9/libs/mips"
             CPPFLAGS+=" -I${ANDROID_NDK_ROOT}/sources/cxx-stl/gnu-libstdc++/4.9/libs/mips/include"
         ;;
+        arm64)
+            LDFLAGS+=" -L${ANDROID_NDK_ROOT}/sources/cxx-stl/gnu-libstdc++/4.9/libs/arm64-v8a"
+            CPPFLAGS+=" -I${ANDROID_NDK_ROOT}/sources/cxx-stl/gnu-libstdc++/4.9/libs/arm64-v8a/include"
+        ;;
         *)
             ANDROID_BUILD_FAIL+=("Unknown combination for ANDROID_BUILD_CXXSTL and TOOLCHAIN_ARCH")
             ANDROID_BUILD_FAIL+=("  ${ANDROID_BUILD_CXXSTL}")
