diff --git a/host/hackrf-tools/src/CMakeLists.txt b/host/hackrf-tools/src/CMakeLists.txt
index 7115151c..a51388ba 100644
--- a/host/hackrf-tools/src/CMakeLists.txt
+++ b/host/hackrf-tools/src/CMakeLists.txt
@@ -23,20 +23,20 @@
 
 set(INSTALL_DEFAULT_BINDIR "bin" CACHE STRING "Appended to CMAKE_INSTALL_PREFIX")
 
-find_package(FFTW REQUIRED)
-include_directories(${FFTW_INCLUDES})
-get_filename_component(FFTW_LIBRARY_DIRS ${FFTW_LIBRARIES} DIRECTORY)
-link_directories(${FFTW_LIBRARY_DIRS})
+#find_package(FFTW REQUIRED)
+#include_directories(${FFTW_INCLUDES})
+#get_filename_component(FFTW_LIBRARY_DIRS ${FFTW_LIBRARIES} DIRECTORY)
+#link_directories(${FFTW_LIBRARY_DIRS})
 
 SET(TOOLS
 	hackrf_transfer
-	hackrf_spiflash
-	hackrf_cpldjtag
+	#hackrf_spiflash
+	#hackrf_cpldjtag
 	hackrf_info
-	hackrf_debug
-	hackrf_clock
-	hackrf_sweep
-	hackrf_operacake
+	#hackrf_debug
+	#hackrf_clock
+	#hackrf_sweep
+	#hackrf_operacake
 )
 
 if(MSVC)
@@ -45,7 +45,7 @@ if(MSVC)
 	)
 	LIST(APPEND TOOLS_LINK_LIBS ${FFTW_LIBRARIES})
 else()
-    LIST(APPEND TOOLS_LINK_LIBS m fftw3f)
+    LIST(APPEND TOOLS_LINK_LIBS m)# fftw3f)
 endif()
 
 if(NOT libhackrf_SOURCE_DIR)
